### PR TITLE
chore(v2): v2 launcher with basic publisher. Fixes #6147

### LINKED ISSUE #6147: v2 launcher with basic publisher

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Fixes #6147 
This PR supports hello world sample -- input parameter placeholder & basic publisher.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: v2/cmd/launcher-v2/main.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	namespace         = flag.String("namespace", "", "The Kubernetes namespace this Pod belongs to.")
+	podName           = flag.String("pod_name", "", "Kubernetes Pod name.")
+	podUID            = flag.String("pod_uid", "", "Kubernetes Pod UID.")
+	pipelineRoot      = flag.String("pipeline_root", "", "The root output directory in which to store output artifacts.")
+	mlmdServerAddress = flag.String("mlmd_server_address", "", "The MLMD gRPC server address.")
+	mlmdServerPort    = flag.String("mlmd_server_port", "8080", "The MLMD gRPC server port.")
+)
+
+func main() {
+	flag.Parse()
```
Comment (capri-xiyue) [inline v2/cmd/launcher-v2/main.go:38]: Do we need to validate the result of the parser?

File: v2/compiler/container.go [hunk 2]
```diff
@@ -111,3 +126,92 @@ func (c *workflowCompiler) addContainerDriverTemplate() string {
...
+				}},
+				ImagePullPolicy: "Always",
+			},
+		}},
+		Container: &k8score.Container{
+			Command: launcherCmd,
+			Args:    userCmdArgs,
+			Image:   container.Image,
+			VolumeMounts: []k8score.VolumeMount{{
+				Name:      "kfp-launcher",
```
Comment (capri-xiyue) [inline v2/compiler/container.go:181]: Do we need to change this string like "kfp-launcher" to constants, I saw it is used in several places.

File: v2/component/launcher_v2.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+			err = fmt.Errorf("failed to create component launcher v2: %w", err)
+		}
+	}()
+	if executionID == 0 {
+		return nil, fmt.Errorf("must specify execution ID")
+	}
+	executorInput := &pipelinespec.ExecutorInput{}
+	err = protojson.Unmarshal([]byte(executorInputJSON), executorInput)
+	if err != nil {
+		return nil, err
```
Comment (capri-xiyue) [inline v2/component/launcher_v2.go:50]: Do we need to add prefix in err msg like "failed to unmarshall xxx"?

File: v2/cmd/launcher-v2/main.go (hunk 1 above)
Comment (Bobgy) [reply v2/cmd/launcher-v2/main.go:38]: Yes, we are already validating them inside the launcher.

File: v2/compiler/container.go (hunk 2 above)
Comment (Bobgy) [reply v2/compiler/container.go:181]: Good point! Updated, I wrote two constants:
* volumePathKFPLauncher
* volumeNameKFPLauncher
they are both "kfp-launcher" at the moment, but they don't necessarily need to be the same

File: v2/component/launcher_v2.go (hunk 3 above)
Comment (Bobgy) [reply v2/component/launcher_v2.go:50]: Good point, the existing error message cannot tell it's for executor input.
Updated

**Comment by Bobgy** [comment]
/retest
go mod download failed randomly, maybe we should add retry

**Comment by Bobgy** [comment]
/retest
strange, kaniko stuck in "taking snapshot of full filesystem"

**Comment by Bobgy** [comment]
I think test infra flakiness is caused by https://github.com/kubeflow/pipelines/issues/6189.
I mitigated by restarting all the nodes again

**Pull request merged**